run_python_examples.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/sh
+#
+# This script runs through the code in each of the python examples.
+# The purpose is just as an integrtion test, not to actually train
+# models in any meaningful way. For that reason, most of these set
+# epochs = 1.
+#
+# Optionally specify a comma separated list of examples to run.
+# can be run as:
+# ./run_python_examples.sh "install_deps,run_all,clean"
+# to pip install dependencies (other than pytorch), run all examples,
+# and remove temporary/changed data files.
+# Expects pytorch to be installed.
+
+BASE_DIR=`pwd`"/"`dirname $0`
+EXAMPLES=`echo $1 | sed -e 's/ //g'`
+
+if which nvcc ; then
+  echo "using cuda"
+  CUDA=1
+  CUDA_FLAG="--cuda"
+else
+  echo "not using cuda"
+  CUDA=0
+  CUDA_FLAG=""
+fi
+
+ERRORS=""
+
+function error() {
+  ERR=$1
+  ERRORS="$ERRORS\n$ERR"
+  echo $ERR
+}
+
+function install_deps() {
+  echo "installing requirements"
+  cat $BASE_DIR/*/requirements.txt | \
+    sort -u | \
+    # testing the installed version of torch, so don't pip install it.
+    grep -vE '^torch$' | \
+    pip install -r /dev/stdin || \
+    { error "failed to install dependencies"; exit 1; }
+}
+
+function start() {
+  EXAMPLE=${FUNCNAME[1]}
+  cd $BASE_DIR/$EXAMPLE
+  echo "Running example: $EXAMPLE"
+}
+
+function dcgan() {
+  start
+  if [ ! -d "lsun" ]; then
+    echo "cloning repo to get lsun dataset"
+    git clone https://github.com/fyu/lsun || { error "couldn't clone lsun repo needed for dcgan";  return; }
```

Comment:
stuff like LSUN and Imagenet are really large datasets.
I wonder if we can do an environment variable like `$DATA_ROOT` to go look for them over there first. This way we can cache / mount them in CI